### PR TITLE
Bump golang to 1.18

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>

# Description of the change(s):
* Bump golang to 1.18

## Why do we need this PR:
*  The upstream hypershift project is using golang 1.18, it is better to upgrade the base image for the hypershift-deployment-controller to 1.18

## Issue reference: 
* https://github.com/stolostron/backlog/issues/22238

## Test API/Unit - Success
```script
ok  	github.com/stolostron/hypershift-deployment-controller/pkg/controllers	22.550s	coverage: 77.7% of statements
ok  	github.com/stolostron/hypershift-deployment-controller/pkg/controllers/autoimport	2.202s	coverage: 61.5% of statements
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -145,7 +148,8 @@ TMP_DIR=$$(mktemp -d) ;\
 cd $$TMP_DIR ;\
 go mod init tmp ;\
 echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+GOBIN=$(PROJECT_DIR)/bin go get -d $(2) ;\
+GOBIN=$(PROJECT_DIR)/bin go install $(3) ;\
```
Comment (zhujian7) [inline Makefile:156]: These can make go-get-tool compatible with 1.17 and 1.18

**Comment by philipwu08** [comment]
@zhujian7 Do we need to change go.mod as well?
```
go 1.17
```

**Review by rokej — APPROVED** [review]

**Comment by zhujian7** [comment]
> @zhujian7 Do we need to change go.mod as well?
> 
> ```
> go 1.17
> ```

@philipwu08  I changed go.mod for hypershift-addon-operator to 1.18 because it needs to build hypershift cli binary. And leave go.mod 1.17 for hypershfit-deployment-controller at present since it can work well with 1.17 and 1.18.

**Pull request merged**